src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Generated/SkillsetsOperationsExtensions.cs

```diff
@@ -0,0 +1,236 @@
+            /// <see href="https://docs.microsoft.com/azure/search/cognitive-search-tutorial-blob" />
+            /// </summary>
+            /// <param name='operations'>
+            /// The operations group for this extension method.
+            /// </param>
+            /// <param name='skillsetName'>
+            /// The name of the skillset to retrieve.
+            /// </param>
+            /// <param name='searchRequestOptions'>
+            /// Additional parameters for the operation
+            /// </param>
+            /// <param name='cancellationToken'>
+            /// The cancellation token.
+            /// </param>
+            public static async Task<Skillset> GetAsync(this ISkillsetsOperations operations, string skillsetName, SearchRequestOptions searchRequestOptions = default(SearchRequestOptions), CancellationToken cancellationToken = default(CancellationToken))
+            {
+                using (var _result = await operations.GetWithHttpMessagesAsync(skillsetName, searchRequestOptions, null, cancellationToken).ConfigureAwait(false))
+                {
+                    return _result.Body;
+                }
+            }
+
+            /// <summary>
+            /// Creates a new Cognitive Search skillset in an Azure Search service.
+            /// <see href="https://docs.microsoft.com/azure/search/cognitive-search-tutorial-blob" />
+            /// </summary>
+            /// <param name='operations'>
+            /// The operations group for this extension method.
+            /// </param>
+            /// <param name='skillsetName'>
+            /// The name of the skillset to create or update.
+            /// </param>
+            /// <param name='skillset'>
+            /// The skillset containing one or more Cognitive Search skills to create or
+            /// update in an Azure Search service.
+            /// </param>
+            /// <param name='searchRequestOptions'>
+            /// Additional parameters for the operation
+            /// </param>
+            public static Skillset CreateOrUpdate(this ISkillsetsOperations operations, string skillsetName, Skillset skillset, SearchRequestOptions searchRequestOptions = default(SearchRequestOptions))
+            {
+                return operations.CreateOrUpdateAsync(skillsetName, skillset, searchRequestOptions).GetAwaiter().GetResult();
+            }
+
+            /// <summary>
+            /// Creates a new Cognitive Search skillset in an Azure Search service.
+            /// <see href="https://docs.microsoft.com/azure/search/cognitive-search-tutorial-blob" />
+            /// </summary>
+            /// <param name='operations'>
+            /// The operations group for this extension method.
+            /// </param>
+            /// <param name='skillsetName'>
+            /// The name of the skillset to create or update.
+            /// </param>
+            /// <param name='skillset'>
```

Comment:
Skillset is required because the operation creates or updates a Skillset (without the Skillset parameter, there's nothing from which to create or update). You're right that having both is redundant though, since the Skillset itself already has a name. I filed an issue on AutoRest years ago to fix this, but after some discussion we decided it's necessary if the code generator needs to handle arbitrary REST APIs, in particular those that support renaming resources. We don't do this, but imagine if you could rename a skillset. To do this, you'd specify the name of the skillset to update for the `skillsetName` parameter, then include the new name you want to use in the `skillset` parameter. Like I said, we don't do this, but in theory some REST APIs might.

I worked around this usability hole with custom code for all the other resources. My custom code just takes the name from the resource and passes it in the separate `name` parameter, effectively hiding it from the customer. We'll need to have AutoRest extended to generate these shortcuts, but until then, we'll need custom code.